recipes/soci/all/conanfile.py

```diff
@@ -0,0 +1,170 @@
+import os
+import copy
+from collections import namedtuple
+from conans import ConanFile, CMake, tools
+
+class SociConan(ConanFile):
+    name = "soci"
+    settings = "os", "arch", "compiler", "build_type"
+    description = "SOCI - The C++ Database Access Library"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "http://soci.sourceforge.net/"
+    license = "BSL-1.0"
+    generators = "cmake"
+    topics = ("conan", "soci", "database")
+    exports_sources=["CMakeLists.txt", "cmake/SOCI-override.cmake"]
+
+    backends = [
+#     "db2",
+      "empty",
+#     "firebird",
+      "mysql",
+      "odbc",
+#     "oracle",
+      "postgresql",
+      "sqlite3"
+    ]
+
+    options = {
+        "shared": [True, False],
+        "with_boost": [True, False],
+        "with_all_backends": [True, False]
+    }
+    options.update({"with_backend_%s" % backend: [True, False, None] for backend in backends})
```

Comment:
```suggestion
    options.update({"with_backend_%s" % backend: [True, False] for backend in backends})
```
You're probably trying to create a tristate, where `None` means `with_all_backends`.
Still, this feels weird and different as other recipes do.